shell/src/main/java/alluxio/cli/bundler/CollectInfo.java

```diff
@@ -19,59 +19,284 @@
+          Option.builder().required(false).longOpt(LOCAL_OPTION_NAME).hasArg(false)
+                  .desc("running only on localhost").build();
+  private static final Options OPTIONS =
+          new Options().addOption(THREAD_NUM_OPTION).addOption(LOCAL_OPTION);
+
+  private static final String TARBALL_NAME = "alluxio-info.tar.gz";
+
+  private ExecutorService mExecutor;
+
+  /**
+   * Creates a new instance of {@link CollectInfo}.
+   *
+   * @param alluxioConf Alluxio configuration
+   */
+  public CollectInfo(InstancedConfiguration alluxioConf) {
+    super(CMD_ALIAS, UNSTABLE_ALIAS, alluxioConf);
+  }
+
+  /**
+   * Finds all hosts in the Alluxio cluster.
+   * We assume the masters and workers cover all the nodes in the cluster.
+   * This command now relies on conf/masters and conf/workers to contain
+   * the nodes in the cluster.
+   * This is the same requirement as bin/alluxio-start.sh.
+   * TODO(jiacheng): phase 2 specify hosts from cmdline
+   * TODO(jiacheng): phase 2 cross-check with the master for which nodes are in the cluster
+   *
+   * @return a set of hostnames in the cluster
+   * */
+  public Set<String> getHosts() {
+    String confDirPath = mConfiguration.get(PropertyKey.CONF_DIR);
+    System.out.format("Looking for masters and workers in %s%n", confDirPath);
+    Set<String> hosts = new HashSet<>();
+    hosts.addAll(CommandUtils.readNodeList(confDirPath, "masters"));
+    hosts.addAll(CommandUtils.readNodeList(confDirPath, "workers"));
+    System.out.format("Found %s hosts%n", hosts.size());
+    return hosts;
+  }
+
   /**
-   * Main method, starts a new CollectInfo.
+   * Main method, starts a new CollectInfo shell.
+   * CollectInfo will SSH to all hosts and invoke {@link CollectInfo} with --local option.
+   * Then collect the tarballs generated on each of the hosts to the localhost.
+   * And tarball all results into one final tarball.
    *
    * @param argv array of arguments given by the user's input from the terminal
    */
   public static void main(String[] argv) throws IOException {
-    int ret = 0;
+    // Parse cmdline args
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+    try {
+      cmd = parser.parse(OPTIONS, argv, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      return;
+    }
+    String[] args = cmd.getArgs();
 
+    // Create the shell instance
     InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
 
-    // Execute the Collectors one by one
-    // Reduce the RPC retry max duration to fall earlier for CLIs
+    // Reduce the RPC retry max duration to fail earlier for CLIs
     conf.set(PropertyKey.USER_RPC_RETRY_MAX_DURATION, "5s", Source.DEFAULT);
     CollectInfo shell = new CollectInfo(conf);
 
-    // If the args are not valid, return early
-    if (argv.length < 2) {
-      System.out.format("Command %s requires at least %s arguments (%s provided)%n",
-              2, argv.length);
```

Comment:
Why is the error output removed? It would be helpful for user to know what exactly is wrong with the argument.

Reply:
My bad, that's an overlook.